examples/ex1.c

```diff
@@ -0,0 +1,68 @@
+
+static char help[] = "Trains binary SVM classification model using precomputed Gramian (linear kernel) matrix.\n\
+                      Provided Gramian matrix is computed and related to no-bias SVM formulation (svm_binary_mod 2).\n";
+
+#include <permonsvm.h>
+
+int main(int argc,char **argv)
+{
+  SVM            svm;
+
+  char           training_file[PETSC_MAX_PATH_LEN] = "data/heart_scale.bin";
+  char           kernel_file[PETSC_MAX_PATH_LEN]   = "data/heart_scale.kernel.bin";
+  char           test_file[PETSC_MAX_PATH_LEN]     = "data/heart_scale.t.bin";
+  PetscBool      test_file_set = PETSC_FALSE;
+
+  PetscViewer    viewer;
+  PetscErrorCode ierr;
+
+  ierr = PermonInitialize(&argc,&argv,(char *)0,help); if (ierr) return ierr;
+
+  ierr = PetscOptionsGetString(NULL,NULL,"-f_training",training_file,sizeof(training_file),NULL);CHKERRQ(ierr);
+  ierr = PetscOptionsGetString(NULL,NULL,"-f_kernel",kernel_file,sizeof(test_file),&test_file_set);CHKERRQ(ierr);
+  ierr = PetscOptionsGetString(NULL,NULL,"-f_test",test_file,sizeof(test_file),&test_file_set);CHKERRQ(ierr);
+
+  /* Create SVM object */
+  ierr = SVMCreate(PETSC_COMM_WORLD,&svm);CHKERRQ(ierr);
+  ierr = SVMSetType(svm,SVM_BINARY);CHKERRQ(ierr);
+  ierr = SVMSetFromOptions(svm);CHKERRQ(ierr);
+
+  /* Load training dataset */
+  ierr = PetscViewerBinaryOpen(PETSC_COMM_WORLD,training_file,FILE_MODE_READ,&viewer);CHKERRQ(ierr);
+  ierr = SVMLoadTrainingDataset(svm,viewer);CHKERRQ(ierr);
+  ierr = PetscViewerDestroy(&viewer);CHKERRQ(ierr);
+
+  /* Load Gramian matrix */
```

Comment:
consider adding
```
if (kernel_file_set) {
```
then this would be the simplest example for all binary type SVM (no complications with different file viewers).

Reply:
Accepted.